NitroxServer/GameLogic/Spawning/EntitySpawner.cs

```diff
@@ -106,5 +110,74 @@ private void SpawnEntities()
                 }
             }
         }
+
+        private bool GetDataFiles(out string lootDistributions, out Dictionary<string, WorldEntityInfo> worldEntityData)
+        {
+            lootDistributions = "";
+            worldEntityData = new Dictionary<string, WorldEntityInfo>();
+            string resourcesPath = "";
+            string gameResourcesPath = Path.Combine(SteamFinder.FindSteamGamePath(264710, "Subnautica"), "Subnautica_Data/resources.assets");
+            AssetsFile resourcesFile;
+            if (File.Exists(gameResourcesPath))
+            {
+                resourcesPath = gameResourcesPath;
+            }
+            else if (File.Exists("../resources.assets"))
+            {
+                resourcesPath = Path.GetFullPath("../resources.assets");
+            }
+            else if (File.Exists("resources.assets"))
+            {
+                resourcesPath = Path.GetFullPath("resources.assets");
+            }
+            else
+            {
+                Log.Error("Could not find resources.assets in parent or current directory.", new FileNotFoundException());
```

Comment:
`throw new FileNotFoundException("Could not find resources.assets in parent or current directory.")`? Then deal with the exception at a higher level rather than passing a bool?

Reply:
Agreed - throwing the exception and reacting to it accordingly in consumer code is a clearer intent that this paradigm.